pkg/prometheus/resource_selector.go

```diff
@@ -192,7 +192,7 @@ func (rs *ResourceSelector) SelectServiceMonitors(ctx context.Context, listFn Li
 		}
 
 		if err = validateScrapeClass(rs.p, sm.Spec.ScrapeClassName); err != nil {
-			rejectFn(sm, fmt.Errorf("scrapeClass: %w", err))
+			rejectFn(sm, err)
```

Comment:
I initially aligned the 3 other resources with this one but @simonpasquier told me the error wrap was not needed so let's align servicemonitor to the others instead

Reply:
sgtm